Assets/Scripts/Environment/Hangar/Battery.cs

```diff
@@ -55,38 +62,77 @@ private void SetEmission()
         meshRenderer.material.SetColor("_EmissionColor", emission);
     }
 
-    public bool PlayerIsHolding()
+    /*public bool PlayerIsHolding()
     {
-        return PlayerManager.PlayerObject.GetComponentInChildren<Battery>() != null;
+        return springJoint != null && springJoint.connectedBody == PlayerManager.PlayerMovement.PlayerRigidbody;
+
+        return PlayerManager.IsHoldingBattery;
     }
+    */
 
     public void TakeBattery()
     {
-        Container.ParentToPlayer();
+        Container.transform.localPosition = new Vector3(Container.transform.localPosition.x, HangarConstants.BatteryYPosition, Container.transform.localPosition.z);
+
+        containerRigidBody.useGravity = false;
+
+        // We add constraints so that the battery doesn't swing around uncontrollably
+        containerRigidBody.constraints = HangarConstants.BatteryRigidbodyConstraints;
+
+        // Setting the container's spring joint values
+        springJoint = Container.AddComponent<SpringJoint>();
+        springJoint.connectedBody = PlayerManager.PlayerObject.GetComponent<Rigidbody>();
+
+        springJoint.spring = HangarConstants.Spring;
+        springJoint.damper = HangarConstants.Damper;
+        springJoint.minDistance = HangarConstants.MinDistance;
+        springJoint.maxDistance = HangarConstants.MaxDistance;
+        springJoint.tolerance = HangarConstants.Tolerance;
+        springJoint.enableCollision = HangarConstants.EnableCollision;
+        springJoint.breakForce = HangarConstants.BreakForce;
+
+        PlayerManager.IsHoldingBattery = true;
+
+        springJointExists = true;
+
     }
 
     public void DropBattery()
     {
         Container.SetParent(HangarManager.BatteriesContainer);
+
+        // As the battery is dropped, we remove the constraint so that the battery can move freely as a physics object
+        containerRigidBody.constraints = RigidbodyConstraints.None;
+        containerRigidBody.useGravity = true;
+        springJointExists = false;
+        PlayerManager.IsHoldingBattery = false;
+        Destroy(springJoint);
     }
 
     private void OnTriggerStay(Collider other)
     {
-        if (IsPlayerColliding && Input.GetKey(PlayerConstants.ActionKey))
+        if (PlayerManager.IsHoldingBattery)
+        {
+            // Don't let the player pick up a battery if they already have one
+            return;
+        }
+
+        else if (IsPlayerColliding && Input.GetKey(PlayerConstants.ActionKey))
         {
-            if (PlayerIsHolding())
-            {
-                // Don't let the player pick up a battery if they already have one
-                return;
-            }
             TakeBattery();
         }
     }
 
     private void Update()
     {
+        // Checking when the spring joint gets destroyed
+        if(springJoint == null && springJointExists == true)

```

Comment:
We should only need to have one 'if' statement for dropping the battery. I think the PlayerIsHolding() is enough. 

When we drop the battery, we are destroying it anyway, so it will always be null.